api/src/main/java/org/openmrs/api/impl/PatientServiceImpl.java

```diff
@@ -593,6 +588,23 @@ public void mergePatients(Patient preferred, Patient notPreferred) throws APIExc
 		Context.getPersonService().savePersonMergeLog(personMergeLog);
 	}
 	
+	private void requireNoActiveOrderOfSameType(Patient patient1, Patient patient2) {
+		String messageKey = "Patient.merge.cannotHaveSameTypeActiveOrders";
+		List<Order> ordersByPatient1 = Context.getOrderService().getAllOrdersByPatient(patient1);
+		List<Order> ordersByPatient2 = Context.getOrderService().getAllOrdersByPatient(patient2);
+		ordersByPatient1.forEach((Order order1) -> {
+			ordersByPatient2.forEach((Order order2) -> {
+				if (order1.isActive() && order2.isActive() && order1.getOrderType().equals(order2.getOrderType())) {
```

Comment:
I'm also commenting based on the conclusion that was made from that discussion. As per the conclusion from that discussion, I don't think the intended behavior is well reflected by the unit tests, it could be because of the way they are named, and the code changes to me don't  seem explicitly address all 3 conditions. Can you try renaming the test methods so that they are more intuitive based on the 3 conditions that need to be met.